### DISCUSSION TITLE
Problems adding NAS source with clean install

### DISCUSSION THREAD

**Discussion by kw-sudo** [rern/rAudio#1353] [General]
Since my first attempt to install r/audio, I have not been able to include the shared music folder on the first try. No matter if in the input mask (+ add) or as cfis-string in fstab. 

The drive is shown with the IP, but with a red status icon. After that there are several - not comprehensible - ways to the goal.

1. edit (save) the cfis-string in fstab by re-inserting the password (doesn't always help)
2. add (+) in the control panel of **the same source** a second time. The source is then displayed twice with green status icon. The funny thing, on the second entry the password is not correct and the character/line length is not identical. If I delete the line with the wrong entry, there is again only one (the first) NAS source with a green icon in the settings (PS: the user data is checked and 100% correct).

If this worked, I create a backup file with this rAudio versioning. If I use this backup file after a new installation, the source settings are immediately applied without errors.

Sometimes, rAudio is really a bit quite funny.

**Comment by kw-sudo** [comment]
Today I did some clean installs and what I can say, the second works for a initial adding (if one can speak of functioning).  

PS: re-mount dosen't work at this point

**Comment by ceejayemm** [comment]
I haven't had any particular problems add a NAS drive to rAudio.  I have used CIFS in the past but not use NFS (from an OpenMediaVaultv6.x server).  After a long discussion with Rern a long time ago I have always:

1. Given the stoirage a name - in my case NAS_Music
2. Always used an IP Number not a name for the server - in my case 192.168.86.250
3. given the full share - in my case /export/music/iTunes/Music (note this is case sensitive from my NFS server)
4. Never specified any optional options

and the mount just happens, time, after time, after time. No need to modify /etc/fstab.

I used a similar setup for CIFS shares but always specified a username/password (as configured on my NAS Samba setup). Again I have not had any particular problems with this setup. In fact I now use a CIFS shared for the Shared Data setup as I want to control the user access more closely which can be achieved with CIFS shares but is (a lot) more difficult with an NFS share.

Chris

**Reply by kw-sudo** [reply]
Except for third, the same settings here. I will try the full path.

PS: but, once it works in my case, it works for long

Oliver

**Reply by kw-sudo** [reply]
Okay, another clean install, editing fstab in the usual way...source has been added, red icon. 

![source2](https://github.com/rern/rAudio/assets/104136223/95e4983a-e1a7-474e-b780-e833fc7e78fd)


Added another (identical) source using the full path to share... the new one flashed red, the original became green at the same time.

![source1](https://github.com/rern/rAudio/assets/104136223/d9f2db51-aa2d-4919-bd97-643f7a0ba64c)

After editing fstab (deleting the wrong one), the original keeps green.

![source2](https://github.com/rern/rAudio/assets/104136223/b92d71e8-fc3b-468f-8206-e0f856c51ef1)

**Reply by rern** [reply]
No need to manually edit fstab.
Click/Tap the plus icon next to Storage.

**Reply by kw-sudo** [reply]
That‘s what I basically did but didn’t work, respectively I came to an identical result (red status icon). 

With my actual „solution“ it also works in that way.

**Reply by kw-sudo** [reply]
@rern Because I'm in the process, the reason for editing fstab is, I get this advice when using the adding (+) workflow.

I can keep repeating this without anything changing. If I go the way via fstab and enter the (also) correct data, I at least get the message that the drive can be found but not mounted. Finishing with my shown 'solution' it works.

![Bildschirmfoto 2023-06-15 um 11 56 27](https://github.com/rern/rAudio/assets/104136223/938a3398-fec6-4c12-989f-ca6c06539fce)

**Reply by kw-sudo** [reply]
@rern 

For a better understanding, I've made a short realtime movie trying to add my NAS source after a clean install. Temporary user data were being checked.

Content:

1. using adding function ('classical way' +) --> error message
2. editing fstab with my cfis string --> source is shown in deactive mode, re-mounting doesn't work
3. adding the same source manually additionally  --> both sources were shown in active mode
4. _if I would delete the redundant source by clicking the 'forget button', both sources would be gone. So..._
5. editing fstab (you see different string length), deleting the second/newer/shorter string, the redundant source disappears, leaves the active

After that, it's works flawless.

[dropbox](https://www.dropbox.com/s/rmo83r4uk9kmcw9/raudio.mp4?dl=0) (9,6MB)

**Reply by cmh714** [reply]
The slashes are messing you up imo. Leave out the leading slash(es)....

**Reply by kw-sudo** [reply]
@cmh714 

At step 3 (adding the same source manually) r/Audio creates the second/new cfis string with exact this leading slashes. As I said, in the end it works.

**EDIT**

If I delete/omit the slashes, I got an UNC error. Adding them again I can re-mount the drive.

**Reply by ceejayemm** [reply]
@master-kw 
I would check the permissions on the server PROVIDING the CIFS share, rAudio requires permissions are shown in the Readme on the main Github page.  I can't say I entirely agree with them as I would prefer to allow the relevant permissions on the CIFS server (in my case OpenMediaVault v6.x) via a valid login and these should not be mandated by the client BUT at the moment rAudio does have specific permission requirements and you need to make sure your CIFS share access setup caters to these to ensure rAudio works correctly.

Not sure I can contribute much more to this thread as I don't personally have the problems described.

Chris

**Reply by kw-sudo** [reply]
Another day, another clean install. I‘ve checked everything and yesterday I could add the NAS at the first run without ‚fstab fumbeling‘.

I‘m fine for the moment. :-)

**Reply by safefly** [reply]
hi master-kw
I believe your video could help me, but does seems to exist any more on dropbox...
with my poor level, I don't succeed do access my Synology (DSM 6.2.4-25556 Update 8) with rAudio rAudio 20250215
Would you help ?
Thanks in advance ;-)

**Comment by markblinkhorn** [comment]
I only use CIFS shares. I always follow the 4 rules given by Chris above. I have *always* had a red icon on the status page but, as the share works anyway, have completely ignored it!

Mark

**Comment by cmh714** [comment]
I only use NFS and its working fine here.
![image](https://github.com/rern/rAudio/assets/8241076/378be628-8651-4a59-be47-310d1fed9fd6)
